jaggr-core/src/main/java/com/ibm/jaggr/core/impl/resource/JsxResourceConverter.java

```diff
@@ -126,7 +171,24 @@ boolean compileJsxResource(final IResource jsxRes, final File target) {
 		if (isTraceLogging) {
 			log.entering(sourceMethod, sourceMethod, new Object[]{jsxRes, target});
 		}
-		boolean result = false;	// not yet implemented
+
+		boolean result;
+		Context ctx = Context.enter();
+		try {
+			// read the contents of the jsx file and convert it
+			String jsx = IOUtils.toString(jsxRes.getInputStream());
+			NativeObject convertedJSX = (NativeObject) transformFunction.call(ctx, globalScope, jsxTransformScript, new String[]{jsx});

```

Comment:
Need to synchronize access to the rhino scope.  This method can be called by multiple, concurrent threads, and rhino scopes are not re-entrant.  Use synchronization to serialize access to the scope, but make sure you don't do any file I/O in the synchronized block.